software/firmware/configuration.py

```diff
@@ -44,6 +44,71 @@ def validate(self, value) -> Validation:
         raise NotImplementedError
 
 
+class FloatConfigPoint(ConfigPoint):
+    """A `ConfigPoint` that requires the selection from a range of floats.  The default value
+    must lie within the specified range
+
+    :param name: The name of this `ConfigPoint`, will be used by scripts to lookup the configured value.
+    :param minimum: The minimum allowed value
+    :param maximum: The maximum allowed value
+    :param default: The default value
+    """
+
+    def __init__(self, name: str, minimum: float, maximum: float, default: float):
+        super().__init__(name=name, type=float, default=default)
+        self.maximum = maximum
+        self.minimum = minimum
+
+        if default < minimum or default > maximum:
+            raise Exception(f"Defaulf {default} is out of range")
+
+    def validate(self, value) -> Validation:
+        """Validates the given value with this ConfigPoint. Returns a `Validation` containing the
+        validation result, as well as an error message containing the reason for a validation failure.
+        """
+        # Silently caste integers into floats, so we don't crash if someone writes e.g. "0" instead of "0.0"
+        if type(value) is int:
+            value = float(value)
+        if type(value) is float:
+            if value >= self.minimum and value <= self.maximum:
+                return VALID
+            else:
+                return Validation(is_valid=False, message=f"Value {value} is out of range")
+        else:
+            return Validation(is_valid=False, message=f"Value {value} is not a number")
+
+
+class IntegerConfigPoint(ConfigPoint):
+    """A `ConfigPoint` that requires selection from a range of integers. The default value
+    must lie within the specified range
+
+    :param name: The name of this `ConfigPoint`, will be used by scripts to lookup the configured value.
+    :param range: The range of valid integers. Note that only the maximum and minimum values are considered;
+                  step values within the range are discarded.
```

Comment:
[discussion] If we're not allowing the step values any longer, there's really no utility to using `range` as part of the interface here. I'd rather switch this to min/max to be consistent with `FloatConfigPoint` than have this note about ignoring the step values of the range.

Reply:
That's fair. I kept the `range` implementation to minimize changes to other files. But I can go through and try to find everywhere that the IntegerConfigPoint is used and update it accordingly.

Reply:
Done. I'm pretty sure I got all of the instances of IntegerConfigPoints being used.